### PR TITLE
fix stco in trailing moov

### PR DESCRIPTION
**Pull request by philippe44**
This is a fix for https://forums.slimdevices.com/forum/user-forums/logitech-media-server/1699393-some-aac-files-result-in-error-decoder-does-not-support-file-format-code-0#post1699393. 

When mp4 files have a trailing moov atom, the stco could be missed and the audio_offset would be wrong. Fix as well the total length check that should be only done once during the first parse

### CODE REVIEW

**Comment by danielvijge** [comment]
I can't say that much about the code quality for a review, but as the original author of the forum post mentioned, I did test this patch, and it does resolve the original issue.

File: Slim/Formats/Movie.pm [hunk 1]
```diff
@@ -252,13 +252,13 @@ sub parseStream {
 	my $offset = $args->{_offset};
 	my $log = logger('player.streaming');
 
-	while (length($args->{_scanbuf}) > $args->{_offset} + $args->{_need} + 8) {
```
Comment (michaelherger) [inline Slim/Formats/Movie.pm:255]: Is this just an optimization, which shouldn't have any impact?...

**Review by michaelherger — APPROVED** [review]
I'll trust your judgement! Should I back port this to 8.5.2?

**Comment by philippe44** [comment]
Yes, absolutely needs to be back ported

File: Slim/Formats/Movie.pm (hunk 1 above)
Comment (philippe44) [reply Slim/Formats/Movie.pm:255]: Kind of - using $offset is just easier to read

**Pull request merged**